core/src/main/java/com/linecorp/armeria/common/NonWrappingRequestContext.java

```diff
@@ -119,7 +119,11 @@ public final SessionProtocol sessionProtocol() {
     @SuppressWarnings("unchecked")
     public <A extends SocketAddress> A remoteAddress() {
         final Channel ch = channel();
-        return ch != null ? (A) ch.remoteAddress() : null;
+        if (ch == null) {
+            return null;
+        }
+        final SocketAddress remoteAddress = ch.remoteAddress();
+        return remoteAddress != null ? (A) remoteAddress : null;
```

Comment:
It seems to me that this change doesn't really change anything. Could you explain why this is better?

Reply:
Yeah, this change brings nothing. Let me revert this, thanks!